dashboard/app/models/script_level.rb

```diff
@@ -744,6 +744,25 @@ def level_deprecated?
     level&.deprecated?
   end
 
+  # WARNING: Do NOT reuse this trashy little method. It is fragile English-only string comparison
+  # written for a very specific use case - logging analytics for the CSA '24-'25 AI Tutor pilot,
+  # in which the level progression naming conventions follow a very specific pattern aligned
+  # with the PRIMM pedagogical approach.
+  #
+  # If the concept of a "progression type" becomes more general, or you're tempted to use this method,
+  # consider a more robust solution such as creating a new property that can be designated by a
+  # Levelbuilder, similar to how we designate assessment levels.
+  def primm_progression_type
+    progression_name = properties["progression"]
+    substring = progression_name.split(":").first if progression_name
+    return "predict_and_run" if substring&.include?("Predict and Run")
```

Comment:
a "Predict and Run level" in this usage is NOT the same as a "Predict Level" like [here](https://github.com/code-dot-org/code-dot-org/blob/b7d50176491461e6c0791d859549be3884612d33/dashboard/app/models/levels/level.rb#L882)